src/amo/reducers/viewContext.js

```diff
@@ -0,0 +1,32 @@
+/* @flow */
+import {
+  ADDON_TYPE_EXTENSION,
+  ADDON_TYPE_THEME,
+  VIEW_CONTEXT_EXPLORE,
+  VIEW_CONTEXT_HOME,
+  SET_VIEW_CONTEXT,
+} from 'core/constants';
+
+
+export type ViewContextType = {|
+  context: ADDON_TYPE_EXTENSION | ADDON_TYPE_THEME | VIEW_CONTEXT_EXPLORE |
```

Comment:
@tofumatt if I'm not wrong, flow doesn't allow (at least currently) to use a `const` as a type, which is unfortunate because it is still possible to make flow able to check this kind of union type, but "by using a more verbose strategy", 
e.g. something similar to the following strategy should work:

- explicitly defining the types of these constants: 

```js
/* @flow */
...

// Define and export the types related to the constants.
export type ViewContextExploreType = 'VIEW_CONTEXT_EXPLORE';
export type ViewContextHomeType = 'VIEW_CONTEXT_HOME';

// Define and export the values related to the constants.
export const VIEW_CONTEXT_EXPLORE: ViewContextExploreType = 'VIEW_CONTEXT_EXPLORE';
export const VIEW_CONTEXT_HOME: ViewContextHomeType = 'VIEW_CONTEXT_HOME';
```

- Import the types related to the constants and define the union using these types:

```js
import type {
  ViewContextHomeType,
  ViewContextExploreType,
  ...
} from './const';

export type ViewContextType = {|
  context: ViewContextHomeType | ViewContextExploreType | ...,
|}
```

This way, flow should be able to check that the following code is not valid because the assigned string does not pass the flow types checks:

```js
var thisShouldBeInvalid: ViewContextType = {context: "this string is not of the valid for this type"};
```